### PR TITLE
OVS-4331 - Speedup DAL

### PR DESCRIPTION
**Pull request by khenderick**
Work in progress

### CODE REVIEW

File: ovs/dal/helpers.py [hunk 1]
```diff
@@ -255,7 +234,7 @@ def check_type(value, required_type):
         elif required_type is float:
             correct = isinstance(value, float) or isinstance(value, int)
             allowed_types = ['float', 'int']
-        elif required_type is int:
+        elif required_type is int or required_type is long:
```
Comment (kvanhijf) [inline ovs/dal/helpers.py:237]: Please keep 'N Sync with check_type in SDM package

File: ovs/dal/helpers.py [hunk 2]
```diff
@@ -274,18 +253,38 @@ def check_type(value, required_type):
...
         """
-        volatile = VolatileFactory.get_client()
-        key = 'ovs_stats_cache_{0}_{1}'.format(cache_type, 'hit' if hit else 'miss')
-        try:
-            successfull = volatile.incr(key)
-            if not successfull:
-                volatile.set(key, 1)
-        except:
-            pass
+        regex = re.compile(r'(\d+)')
```
Comment (kvanhijf) [inline ovs/dal/helpers.py:262]: Inner brackets are obsolete: `re.compile(r'\d+')`

File: ovs/dal/hybrids/t_testdisk.py [hunk 3]
```diff
@@ -36,7 +36,8 @@ class TestDisk(DataObject):
                    Relation('parent', None, 'children', mandatory=False)]
     __dynamics = [Dynamic('used_size', int, 5),
                   Dynamic('wrong_type', int, 5),
-                  Dynamic('updatable', int, 5)]
+                  Dynamic('updatable', int, 5),
```
Comment (kvanhijf) [inline ovs/dal/hybrids/t_testdisk.py:39]: Updatable is written incorrectly and obsolete anyway

File: ovs/dal/lists/mgmtcenterlist.py [hunk 4]
```diff
@@ -30,21 +29,16 @@ def get_mgmtcenters():
...
         """
-        mgmtcenters = DataList({'object': MgmtCenter,
-                                'data': DataList.select.GUIDS,
-                                'query': {'type': DataList.where_operator.AND,
-                                          'items': [('ip', DataList.operator.EQUALS, ip)]}}).data
-        if mgmtcenters:
-            return DataObjectList(mgmtcenters, MgmtCenter)[0]
+        mgmtcenters = DataList(MgmtCenter, {'type': DataList.where_operator.AND,
+                                            'items': [('ip', DataList.operator.EQUALS, ip)]})
+        if len(mgmtcenters):
```
Comment (kvanhijf) [inline ovs/dal/lists/mgmtcenterlist.py]: `> 0`

File: ovs/dal/lists/pmachinelist.py [hunk 5]
```diff
@@ -31,23 +30,18 @@ def get_pmachines():
...
         """
-        pmachines = DataList({'object': PMachine,
-                              'data': DataList.select.GUIDS,
-                              'query': {'type': DataList.where_operator.AND,
-                                        'items': [('ip', DataList.operator.EQUALS, ip)]}}).data
-        if pmachines:
-            return DataObjectList(pmachines, PMachine)[0]
+        pmachines = DataList(PMachine, {'type': DataList.where_operator.AND,
+                                        'items': [('ip', DataList.operator.EQUALS, ip)]})
+        if len(pmachines):
```
Comment (kvanhijf) [inline ovs/dal/lists/pmachinelist.py]: `> 0`

File: ovs/dal/lists/vdisklist.py [hunk 6]
```diff
@@ -56,14 +48,10 @@ def get_vdisk_by_name(vdiskname):
...
         """
-        # pylint: disable=line-too-long
-        vdisks = DataList({'object': VDisk,
-                           'data': DataList.select.GUIDS,
-                           'query': {'type': DataList.where_operator.AND,
-                                     'items': [('name', DataList.operator.EQUALS, vdiskname)]}}).data
-        # pylint: enable=line-too-long
+        vdisks = DataList(VDisk, {'type': DataList.where_operator.AND,
+                                  'items': [('name', DataList.operator.EQUALS, vdiskname)]})
         if vdisks:
```
Comment (kvanhijf) [inline ovs/dal/lists/vdisklist.py]: `if len(vdisks) > 0`

File: ovs/dal/lists/vdisklist.py [hunk 7]
```diff
@@ -71,17 +59,13 @@ def get_by_devicename_and_vpool(devicename, vpool):
...
-        vds = DataList({'object': VDisk,
-                        'data': DataList.select.GUIDS,
-                        'query': {'type': DataList.where_operator.AND,
-                                  'items': [('devicename', DataList.operator.EQUALS, devicename),
-                                            ('vpool_guid', DataList.operator.EQUALS, vpool.guid)]}}).data  # noqa
-        # pylint: enable=line-too-long
+        vds = DataList(VDisk, {'type': DataList.where_operator.AND,
+                               'items': [('devicename', DataList.operator.EQUALS, devicename),
+                                         ('vpool_guid', DataList.operator.EQUALS, vpool.guid)]})
         if vds:
```
Comment (kvanhijf) [inline ovs/dal/lists/vdisklist.py]: `if len(vds) > 0:`
and many more

File: ovs/dal/datalist.py [hunk 8]
```diff
@@ -62,193 +56,193 @@ class Operator(object):
...
-            if isinstance(item, dict):
-                # Recursive
-                if item['type'] == DataList.where_operator.AND:
-                    result = self._exec_and(instance, item['items'])
-                else:
-                    result = self._exec_or(instance, item['items'])
-                if result is True:
-                    return True
+                if item['type'] not in [DataList.where_operator.AND, DataList.where_operator.OR]:
+                    raise NotImplementedError('Invalid where operator specified')
```
Comment (khenderick) [inline ovs/dal/datalist.py]: Check could be at the start of the `_filter`

File: ovs/dal/helpers.py (hunk 1 above)
Comment (khenderick) [inline ovs/dal/helpers.py:237]: Ok

File: ovs/dal/helpers.py (hunk 2 above)
Comment (khenderick) [inline ovs/dal/helpers.py:262]: They must be in there.

File: ovs/dal/hybrids/t_testdisk.py (hunk 3 above)
Comment (khenderick) [inline ovs/dal/hybrids/t_testdisk.py:39]: Updatable is actually used below, and updatable is actually more frequently used compared to updateable.

**Pull request merged**